lib/shared_page_pool.ml

```diff
@@ -92,7 +92,7 @@ let rec use t fn =
       let b2 = Cstruct.shift page block_size in
       let id1 = t.next_id in
       let id2 = t.next_id + 1 in
-      t.next_id <- t.next_id + 2;
+      t.next_id <- (t.next_id + 2) mod (1 lsl 16);
```

Comment:
This can't wrap, because we already compared against `max_pages` above. If it does wrap, we'd certainly have two pages in the pool with the same ID, which is what we're trying to avoid.